extensions/ql-vscode/src/extension.ts

```diff
@@ -493,6 +494,32 @@ async function activateWithInstalledDistribution(
     }
   }
 
+  async function previewQueryHelp(
+    selectedQuery: Uri
+  ): Promise<void> {
+    const pathToQhelp = selectedQuery ? selectedQuery.fsPath : window.activeTextEditor!.document.uri.fsPath;
+    if(pathToQhelp) {
+      // Create temporary directory
+      return tmp.dir().then(async o => {
+        const relativePathToMd = path.basename(pathToQhelp, '.qhelp') + '.md';
+        const absolutePathToMd = path.join(o.path, relativePathToMd);
+        const uri = Uri.file(absolutePathToMd);
+        try {
+          await cliServer.generateQueryHelp(pathToQhelp , absolutePathToMd);
+          await commands.executeCommand('markdown.showPreviewToSide', uri);
+        } catch (err) {
+          const errorMessage =  err.message.includes('Generating qhelp in markdown') ? (
+            `Could not generate markdown from ${pathToQhelp}: Bad formatting in .qhelp file.`
+          ) : `Could not open a preview of the generated file (${absolutePathToMd}).`;
+          void helpers.showAndLogErrorMessage(errorMessage);
```

Comment:
As @adityasharad mentioned above (in https://github.com/github/vscode-codeql/pull/988#discussion_r740549449), it would be nice to print `err` itself somewhere (perhaps using `logger.log(...)`?) 📝